distributed/scheduler.py

```diff
@@ -2213,7 +2256,9 @@ def transition_released_waiting(self, key):
                 pdb.set_trace()
             raise
 
-    def transition_no_worker_waiting(self, key):
+    @ccall
```

Comment:
why all `ccall` rather than `cfunc` - can they be called from anywhere but `_transition`?